File: frontend/src/types/escalation.ts
export type EscalationTier = 'TIER_0_NORMAL' | 'TIER_1_VISUAL' | 'TIER_2_PUSH' | 'TIER_3_DISPATCH' | 'ACKNOWLEDGED';

export interface AlertEscalationStatus {
  case_id: string;
  case_code: string;
  hospital_id: string;
  hospital_name: string;
  time_alerted: string;
  seconds_elapsed: number;
  escalation_tier: EscalationTier;
  is_escalated: boolean;
  requires_audible_chime: boolean;
  requires_dispatch_alert: boolean;
  acknowledged: boolean;
}

export interface HospitalAlertEscalationsResponse {
  hospital_id: string;
  hospital_name: string;
  total_unacknowledged_alerts: number;
  max_escalation_tier: EscalationTier;
  escalations: AlertEscalationStatus[];
}

File: backend/app/schemas/escalation.py
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field


class AlertEscalationStatus(BaseModel):
    """
    Status of an unacknowledged emergency alert for a specific hospital case.
    """
    case_id: str
    hospital_id: str
    alert_timestamp: datetime
    elapsed_seconds: float
    escalation_tier: str  # TIER_0_NORMAL, TIER_1_VISUAL, TIER_2_PUSH, TIER_3_DISPATCH
    tier_label: str
    is_escalated: bool
    requires_audible_chime: bool
    requires_dispatch_alert: bool
    acknowledged: bool


class HospitalAlertEscalationsResponse(BaseModel):
    """
    Summary of all active alert escalations for a hospital emergency department.
    """
    hospital_id: str
    timestamp: datetime
    total_unacknowledged_alerts: int
    escalated_count: int
    alerts: List[AlertEscalationStatus]
